client/src/pages/EditProject.js

```diff
@@ -0,0 +1,327 @@
+    width: '100%',
+    marginTop: '30px'
+  },
+  dateField: {
+
+  },
+  formLine: {
+    marginBottom: '20',
+  },
+  button: {
+    margin: '20px 0px 10px 0px'
+  }
+}
+
+class EditProject extends Component {
+  constructor(props) {
+    super(props);
+    this.validators = FormValidator([
+      {
+        field: 'title',
+        method: validator.isEmpty,
+        validWhen: false,
+        message: 'Title is required.'
+      },
+      {
+        field: 'industry',
+        method: validator.isEmpty,
+        validWhen: false,
+        message: 'Industry is required'
+      },
+      {
+        field: 'location',
+        method: validator.isEmpty,
+        validWhen: false,
+        message: 'location is required'
+      },
+      {
+        field: 'fundingGoal',
+        method: validator.isNumeric,
+        validWhen: true,
+      },
+      {
+        field: 'fundingDeadline',
+        method: (value) => validator.isAfter(value),
+        validWhen: true,
+      }
+    ]);
+    this.state = {
+      project: {
+        projectId: null,
+        title: '',
+        subtitle: '',
+        description: '',
+        industry: '',
+        location: '',
+        updatedImages: [],
+        fundingGoal: 0,
+        fundingDeadline: ''
+      },
+      formData: {},
+      projectUserId: null,
+      validation: this.validators.valid(),
+      uploading: false
+    }
+  }
+
+  componentDidMount() {
+    const { id: projectId, title, subtitle, description, industry, location, images: updatedImages, fundingGoal, projectUserId } = this.props.location.state
+    const fundingDeadline = moment(this.props.location.state.fundingDeadline).format('YYYY-MM-DD');
+    const project = { projectId, title, subtitle, description, industry, location, updatedImages, fundingGoal, fundingDeadline }
+    this.setState({ project, projectUserId, formData: new FormData() })
+  }
+
+  handleInput = (event) => {
+    const { value, name } = event.target;
+    const { project } = this.state;
+    project[name] = value;
+    this.setState({ project })
+  }
+
+  handleSubmit = async (event) => {
+    event.preventDefault();
+    try {
+      const validation = this.validators.validate(this.state.project);
+      this.setState({ validation });
+
+      const { project, projectUserId, formData } = this.state;
+
+      if (validation.isValid) {
+        // for (const name in project) { // have an odd and interesting error where an `ignore_whitespace` property is added with the value of false
+        //   console.log(name);
+        //   formData.set(name, project[name]);
+        // }
+        formData.set('title', project.title);
+        formData.set('subtitle', project.subtitle);
+        formData.set('description', project.description);
+        formData.set('industry', project.industry);
+        formData.set('location', project.location);
+        formData.set('fundingGoal', project.fundingGoal);
+        formData.set('fundingDeadline', project.fundingDeadline);
```

Comment:
there must be a more convenient helper function we can do here, but not too important to block it

Reply:
That was actually what the for .. in loop was supposed to handle in the lines above but I kept getting this weird bug with "ignore_whitespace" and I could not figure out where it was coming from

Reply:
strange

Reply:
oh! @DevDelvo might want to try for ... of ... not for ... in ... ignore_whitespace might be a prototype property